### PR TITLE
Workaround for KEYCLOAK-11282

### PR DESCRIPTION
**Pull request by gytis**
Fixes https://github.com/snowdrop/secured-example/issues/67

### CODE REVIEW

**Comment by gytis** [comment]
I had to disable the trust manager, because since the starter upgrade this example stopped working with an authentication server deployed on OpenShift which has self-signed certificate.

**Comment by gytis** [comment]
I'll push the same commit with trust manager disabling to `sb-2.2.x` branch. That branch doesn't need a workaround for KEYCLOAK-11282 because it uses a newer starter version which has a fix

**Comment by gytis** [comment]
I'm looking into the CI failure now

**Comment by gytis** [comment]
If ran with `spring-boot:run` the application authenticates as expected. But integration test which uses different authentication client - fails. I'm looking into its configuration

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -2,6 +2,8 @@ keycloak.realm=${realm:master}
 keycloak.realm-key=${realm.public.key:MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAoETnPmN55xBJjRzN/cs30OzJ9olkteLVNRjzdTxFOyRtS2ovDfzdhhO9XzUcTMbIsCOAZtSt8K+6yvBXypOSYvI75EUdypmkcK1KoptqY5KEBQ1KwhWuP7IWQ0fshUwD6jI1QWDfGxfM/h34FvEn/0tJ71xN2P8TI2YanwuDZgosdobx/PAvlGREBGuk4BgmexTOkAdnFxIUQcCkiEZ2C41uCrxiS4CEe5OX91aK9HKZV4ZJX6vnqMHmdDnsMdO+UFtxOBYZio+a1jP4W3d7J5fGeiOaXjQCOpivKnP2yU2DPdWmDMyVb67l8DRA+jh0OJFKZ5H2fNgE3II59vdsRwIDAQAB}
 keycloak.auth-server-url=${SSO_AUTH_SERVER_URL}
 keycloak.resource=${client.id:secured-example-endpoint}
+# For example purposes only. Never disable trust manager in production
```
Comment (geoand) [inline src/main/resources/application.properties:5]: :+1:

**Review by geoand — APPROVED** [review]

**Comment by gytis** [comment]
I was able to reproduce the failure on my machine initially. Then after 5ish runs trying to debug different bits it started to pass without any changes and it keeps passing locally. But it still fails in CI...

**Comment by geoand** [comment]
:cry:

**Comment by gytis** [comment]
@cmoulliard @metacosm @geoand I suggest that we merge this PR if you agree with the changes while ignoring the CI failures. I will create a follow-up issue to deal with CircleCI which will need a much bigger consideration on what to do. In a meantime these code changes work with Minishift and Hetzner.

**Comment by geoand** [comment]
Fine with me

**Comment by cmoulliard** [comment]
LGTM.  Why should we continue to test it on okd instead of deploying it on k8s using kind as tool to bootstrap the cluster ?

**Comment by gytis** [comment]
That's how it was setup since the beginning. The examples use routes and other OpenShift features, so I assume we need OKD to test that. But if we're allowed to use Kubernetes and to simplify our setup I'm all for it

**Comment by cmoulliard** [comment]
Openshift tests could only be handled by QE and use k8s for OSS.

**Comment by gytis** [comment]
I don't think that QE has separate tests for our examples. I think they only run what's in this repo which is OpenShiftIT.java

**Comment by gytis** [comment]
But I agree that your proposition makes sense @cmoulliard. Maybe we should discuss about this during the weekly call.

**Comment by gytis** [comment]
Maintaining OpenShift tests in an opensource CI isn't an easy task.

**Pull request merged**